cli/pkg/kctrl/cmd/package/builder/create.go

```diff
@@ -0,0 +1,450 @@
+	"github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/package/builder/fetch"
+	"github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/package/builder/template"
+	"github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/logger"
+	"github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/util"
+	kappctrlapis "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/kappctrl/v1alpha1"
+	"github.com/vmware-tanzu/carvel-kapp-controller/pkg/apiserver/apis/datapackaging/v1alpha1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/yaml"
+)
+
+type CreateOptions struct {
+	ui          ui.UI
+	depsFactory cmdcore.DepsFactory
+	logger      logger.Logger
+
+	NamespaceFlags cmdcore.NamespaceFlags
+	Name           string
+
+	ValuesSchema      bool
+	DefaultValuesFile string
+
+	pkgCmdTreeOpts cmdcore.PackageCommandTreeOpts
+}
+
+func NewCreateOptions(ui ui.UI, logger logger.Logger, pkgCmdTreeOpts cmdcore.PackageCommandTreeOpts) *CreateOptions {
+	return &CreateOptions{ui: ui, logger: logger, pkgCmdTreeOpts: pkgCmdTreeOpts}
+}
+
+func NewCreateCmd(o *CreateOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "create",
+		Aliases: []string{"g"},
+		Short:   "Create a package",
+		Args:    cobra.ExactArgs(1),
+		RunE:    func(_ *cobra.Command, args []string) error { return o.Run(args) },
+		Example: cmdcore.Examples{
+			cmdcore.Example{"Create a package",
+				[]string{"package", "builder", "create"},
+			},
+		}.Description("", o.pkgCmdTreeOpts),
+		SilenceUsage: true,
+		Annotations:  map[string]string{"table": ""},
+	}
+
+	return cmd
+}
+
+func (o *CreateOptions) Run(args []string) error {
+	//TODO Rohit Should we provide an option to give pkg location?
+	pkgLocation := GetPkgLocation()
+	createStep := NewCreateStep(o.ui, pkgLocation)
+	err := createStep.Run()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+type CreateStep struct {
+	Ui           ui.UI
+	PkgLocation  string
+	PkgVersion   string
+	FqName       string
+	ValuesSchema v1alpha1.ValuesSchema
+	Fetch        fetch.FetchStep
+	Template     template.TemplateStep
+	Maintainers  []v1alpha1.Maintainer
+}
+
+func NewCreateStep(ui ui.UI, pkgLocation string) *CreateStep {
+	return &CreateStep{
+		Ui:          ui,
+		PkgLocation: pkgLocation,
+	}
+}
+
+func (createStep *CreateStep) Run() error {
+	err := createStep.PreInteract()
+	if err != nil {
+		return err
+	}
+	err = createStep.Interact()
+	if err != nil {
+		return err
+	}
+	err = createStep.PostInteract()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (createStep CreateStep) getStartBlock() []byte {
+	str := fmt.Sprintf(`
+# Lets start on the package creation process.
+# First we need a directory to store all configurations
+# Creating directory %s
+# 	$ mkdir -p %s
+`, createStep.PkgLocation, createStep.PkgLocation)
+	return []byte(str)
+}
+
+func (create CreateStep) PreInteract() error {
+	create.Ui.PrintBlock(create.getStartBlock())
```

Comment:
should not be using PrintBlock for most of the text, use BeginLinef

Reply:
Now using the PrintBlock only when we are cat'ing the files such as vendir.yml, package.yml and packageMetadata.yml. Apart from this, everywhere using BeginLinef